src/pcrlock/pcrlock.c

```diff
@@ -4567,15 +4567,36 @@ static int make_policy(bool force, bool recovery_pin) {
                 log_info("Retrieved PIN from TPM2 in %s.", FORMAT_TIMESPAN(usec_sub_unsigned(now(CLOCK_MONOTONIC), pin_start_usec), 1));
         }
 
-        TPM2B_NV_PUBLIC nv_public = {};
+        /* Load the PIN into the TPM so that we can use it in a TPM2_PolicyDigest() to write to the nvindex */
+        _cleanup_(tpm2_handle_freep) Tpm2Handle *pin_handle = NULL;
+        r = tpm2_seal_data(
+                        tc,
+                        &IOVEC_MAKE(auth.buffer, auth.size),
+                        srk_handle,
+                        encryption_session,
+                        /* policy_digest= */ NULL,
+                        /* ret_public= */ NULL,
+                        /* ret_private= */ NULL,
+                        &pin_handle);
```

Comment:
> TPM2_PolicySigned doesn't actually require the object to have the TPMA_OBJECT_SIGN attribute set

It's validating the signature that could be external to the TPM, so the attributes would be meaningless. See the example in the tools manpage
 - https://github.com/tpm2-software/tpm2-tools/blob/f2adc763577fc80f502bef748ac6da5f8e066aae/man/tpm2_policysigned.1.md?plain=1#L2

> TPM2_VerifySignature, behaves differently here, as it requires that attribute which means it only accepts an object that is a HMAC key. I wonder if this is an accidental inconsistency in the TPM2 spec.

The key here is just a keyedhash key, setup the same as a "sealed" object. It is not an HMAC key afaict. An HMAC key would have the scheme set in the `TPMT_PUBLIC`.

@poettering without knowing exactly what the problem is, this may be the issue, you need to set the keyedhash scheme to hmac and the algorithm to sha256. Use tpm2_create -Ghmac:sha256 for an example of what the public should be populated as.